src/main/java/httpserver/HttpRouter.java

```diff
@@ -1,22 +1,22 @@
 package httpserver;
 
-public class HttpRouter implements Router {
+import java.util.Map;
+
+public class HttpRouter implements Routeable {
 
-    private final Action getWithBody;
-    private final Action getWithoutBody;
     private final Action methodNotAllowed;
+    private final Map<String, Map<String, Action>> routes;
 
-    public HttpRouter(Action getWithBody, Action getWithoutBody, Action methodNotAllowed) {
-        this.getWithBody = getWithBody;
-        this.getWithoutBody = getWithoutBody;
+    public HttpRouter(Map<String, Map<String, Action>> routes, Action methodNotAllowed) {
```

Comment:
We have this really nice object that encapsulates building up the routes, but we break that encapsulation when we pass in a Map here. What if we passed in `Routes` instead?

Reply:
Sure we can do that